### ISSUE TITLE
Support permission of invitee only

### ISSUE THREAD

**Issue by jackycute** [hackmdio/hackmd#35]
Should support a permission of invitee only.
Only the invitee from email or link can edit the note.

**Comment by jackycute** [comment]
We supported `private` permission in 49c7dded4539044d8053dba8d3fe24b97056c0d2.
`private` means only owner can view and edit.

**Comment by jackycute** [comment]
We supported `limied` and `protected` permissions in a8068d38d589a1b8ed008f737d42ed35df4bfd2b.
`limited` is similar to `editable` but forbid guest to view.
`protected` is similar to `locked` but forbid guest to view.

**Comment by ccoenen** [comment]
This list is quite long by now. While I find all the items' names to be very carefully chosen, yet I still have to look up which is which. While permissions are being extended, maybe it's a good idea to revisit how a user sets these permissions. Maybe it's not one long list from "very private" to "very public" but two sliders, one for reading and one for writing. Or something else entirely.

**Comment by jackycute** [comment]
@ccoenen 
So actually we only need to choose one of `read` or `write` permission for every invited user.
Not one from current permission types.

**Comment by Himura2la** [comment]
I'm not quite understand your permission system. If I set `limited`, does it mean that everyone who is registered on the site can view/edit it?
How can I allow access only to the list of certain persons?

**Comment by jackycute** [comment]
There is no way to allow only for certain people to access note for now.
So this is the issue open for it.

We have six permissions for now, I would suggest to read it here: https://hackmd.io/features#permissions

**Comment by Himura2la** [comment]
Thanks, looking forward for having this feature implemented.

**Comment by Keeo** [comment]
Let me add a little bump to this issue. Sadly this missing feature forced me to drop this tool in favor of something else.

**Comment by ccoenen** [comment]
@Keeo Out of curiosity: what alternative are you using?

**Comment by Keeo** [comment]
@ccoenen We downloaded markdown file from hackmd and placed it inside github repository. Collaboration is done thought normal github process and just for rendering use hackmd.

**Comment by illyas3** [comment]
One of the best editors out there, yet one of the worst permission systems. How is this even reasonable to let anyone signed in to edit your notes?

**Comment by SISheogorath** [comment]
You can change the permission to freely, when you don't want everyone to sign in.

The permission system is mainly grown over time. It definitely needs re-factoring but that also needs time :) If you want to spend some time on the project and work on the permission system, that's definitely an appreciated work.

So feel free to file a PR :)

If you have question or want to talk about concepts feel free to join the Matrix and Gitter community and talk with us.

**Comment by ccoenen** [comment]
It would certainly rock to have something like
"this list of people is allowed"
or
"these five people form a group, that group owns this document"
or
"i can edit, only that group can read"

but somebody has to write that, and currently even something as basic as user accounts is, let's say, very rudimentary. To be honest, I wouldn't expect this any time soon unless somebody picks it up as a pull request.

**Comment by gandhiano** [comment]
Why not an invite-only link with read and/or write permissions, like it is conventional in many other web tools (e.g. nextcloud)? It is so, that very often there is the need to share documents to review with external people who won't additionally create an account. 

They could via such a link just jump in to the document and start editing. And this wouldn't really require an additional permission in the list (I also find they start getting very confusing, as someone commented above), just an option to share a link, e.g. via a button next to Publish.

**Comment by jaragunde** [comment]
I think that the password feature (#494) could be a workaround for most use cases mentioned here, without making the permission system more complicated.

**Comment by SISheogorath** [comment]
No, no changes on that, yet. https://demo.codimd.org/features#Permissions

But we welcome contributions! If you want to help or just talk a bit about it, feel free to join us in our Matrix community: [`#codimd:matrix.org`](https://riot.im/app/#/room/#codimd:matrix.org)

**Comment by abhijitm08** [comment]
Hi All, This feature would be very nice to have. Would it be possible to bump this up in priority?
